components/builder-worker/src/runner/mod.rs

```diff
@@ -0,0 +1,337 @@
+            }
+        };
+        if !self.post_process(&mut archive) {
+            // JW TODO: We should shelve the built artifacts and allow a retry on post-processing.
+            // If the job is killed then we can kill the shelved artifacts.
+            return self.fail();
+        }
+        self.complete()
+    }
+
+    pub fn build(&self, logger: &mut Logger) -> Result<PackageArchive> {
+        let args =
+            vec![OsString::from("-s"),
+                 OsString::from(self.workspace.src()),
+                 OsString::from("-r"),
+                 OsString::from(self.workspace.studio()),
+                 OsString::from("-k"),
+                 OsString::from("core"),
+                 OsString::from("build"),
+                 OsString::from(Path::new(self.workspace.job.get_project().get_plan_path())
+                     .parent()
+                     .unwrap())];
+        let command = match PackageInstall::load(&STUDIO_PKG, None) {
+            Ok(package) => format!("{}/hab-studio", package.paths().unwrap()[0].display()),
+            Err(_) => panic!("should install core/hab-studio because it's missing?"),
+        };
+        debug!("building, cmd={:?}, args={:?}", command, args);
+        let mut child = Command::new(command)
+            .args(&args)
+            .env_clear()
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to spawn child");
+        if let Some(ref mut stdout) = child.stdout {
+            for line in BufReader::new(stdout).lines() {
+                let mut l: String = line.unwrap();
+                let eol: &str = "\n";
+                l = l + eol;
+                logger.log(l.as_bytes());
+            }
+        }
+        let exit_status = child.wait().expect("failed to wait on child");
+        debug!("build complete, status={:?}", exit_status);
+        if exit_status.success() {
+            self.workspace.last_built()
+        } else {
+            Err(Error::BuildFailure(exit_status.code().unwrap_or(-1)))
+        }
+    }
+
+    pub fn job(&self) -> &proto::Job {
+        &self.workspace.job
+    }
+
+    fn complete(&mut self) -> () {
+        self.teardown().err().map(|e| error!("{}", e));
+        self.workspace.job.set_state(proto::JobState::Complete);
+    }
+
+    fn fail(&mut self) -> () {
+        self.workspace.job.set_state(proto::JobState::Failed);
+    }
+
+    fn post_process(&self, archive: &mut PackageArchive) -> bool {
+        // JW TODO: In the future we'll support multiple and configurable post processors, but for
+        // now let's just publish to the public depot
+        //
+        // Things to solve right now
+        // * Where do we get the token for authentication?
+        //      * Should the workers ask for a lease from the JobSrv?
+        let client =
+            depot_client::Client::new("https://app.habitat.sh/v1/depot", PRODUCT, VERSION, None)
```

Comment:
You should be able to get the URL from [`core::url::DEFAULT_DEPOT_URL`](https://github.com/habitat-sh/habitat/blob/ddf8823a820492457cab56ac0f727089c94f7fd5/components/core/src/url.rs#L18) or calling [`core::url::default_depot_url()`](https://github.com/habitat-sh/habitat/blob/ddf8823a820492457cab56ac0f727089c94f7fd5/components/core/src/url.rs#L23-L28) which honors the `HAB_DEPOT_URL` environment variable. That function could be the mechanism by which we could upload into a channel or alternative Depot (for the moment, by environment variable).